### PR TITLE
SOFTWARE-3919 Add documentation and architecture design

### PR DESCRIPTION
**Pull request by efajardo**
[SOFTWARE-3919](https://opensciencegrid.atlassian.net/browse/SOFTWARE-3919)

### CODE REVIEW

File: README [hunk 1]
```diff
@@ -1,44 +1,35 @@
...
-To start:
-
-rsv-control --enable org.osg.local-gfactory-site-querying-local --host host_name
-
-rsv-control --on org.osg.local-gfactory-site-querying-local --host host_name
\ No newline at end of file
+Use Case
+========
+Rsv-gwms-tester is a package that is meant to do end-to-end regular testing of [GlideinWMS](https://glideinwms.fnal.gov/).
+More specifically it is mean to submit a set of HTCondor Vanila universe jobs to the set of supported grid sites of a Virtual Organization (VO).
```
Comment (brianhlin) [inline README]:
```suggestion
More specifically it is mean to submit a set of HTCondor Vanilla universe jobs to the set of supported grid sites of a Virtual Organization (VO).
```

File: README [hunk 2]
```diff
@@ -1,44 +1,35 @@
...
+It can be thought of as a [SAM-light](https://twiki.cern.ch/twiki/bin/view/CMSPublic/CompOpsSAMTests) set of grid tests.
+
+Requirements
+============
+1. The Virtual Organization has already a working [Glideinwms Frontend](https://glideinwms.fnal.gov/doc.prd/frontend/index.html)
+1. The Virtual Organization creates a set of probes(jobs or capabilities it wants to test out of a site).
+1. A Rhel6/6 machine that can install rsv as root.
```
Comment (brianhlin) [inline README]:
```suggestion
1. An EL6 machine that can install rsv as root.
```

Why the EL6 requirement?

File: README [hunk 3]
```diff
@@ -1,44 +1,35 @@
...
+1. A Public IP
+
+Architecture
+============
+The architechture of `rsv-gwms-tester` is based on the architechture of RSV. It is meant as a set of probes (jobs)
+that are periodically run using HTCondor-cron.
+Traditionally RSV installations would submit grid tests(HTCondor-C) from a host to the same host HTCondor-CE.
+`rsv-gwms-tester` submits a set of Vanilla Jobs that are meant to run at different locations in the Grid.
+
+The basic idea is that a user (A VO Manager) just needs to edit one "master" probe's configuration (org.osg.local-gfactory-querying-local) select the VO it wants to test.
```
Comment (brianhlin) [inline README]:
```suggestion
The basic idea is that a user (A VO Manager) just needs to edit one "master" probe's configuration (org.osg.local-gfactory-querying-local) and select the VO it wants to test.
```

File: README [hunk 4]
```diff
@@ -1,44 +1,35 @@
...
+Then the master probes will enable the actual "test probes" at each site the VO is allowed to run in.
+It does by querying a factory HTCondor collector and filtering the sites based on if they suppport a given organization,
+or the sites match a specific constraint (i.e Give me the name of sites that suport LIGO and VIRGO).
+From then on the submit host where `rsv-gwms-tester` runs is just like any other submit host in a HTCondor-Pool.
```
Comment (brianhlin) [inline README]:
```suggestion
From then on the submit host where `rsv-gwms-tester` runs is just like any other submit host in a HTCondor pool.
```

File: README [hunk 5]
```diff
@@ -1,44 +1,35 @@
...
+
+Behind the scenes the master probes enables the same test jobs for each site it finds on the list obtained above.
+Each job is a probe and while the default configuration ships a "dummy-vanilla-probe" this is meant as a template for each organization to write downs as many probes for what they want to test a site(i.e software installation, access to storage or network setup).
```
Comment (brianhlin) [inline README]:
```suggestion
Each job is a probe and while the default configuration ships a "dummy-vanilla-probe" this is meant as a template for each organization to write downs as many probes for what they want to test a site (i.e software installation, access to storage or network setup).
```

**Review by brianhlin — APPROVED** [review]
A few minor suggestions but otherwise these changes are fine. One thing that's not immediately clear to me: are these like "light pilots" or are they payloads that run inside the pilot? 

Also, in the future, please use a 120-char line limit. It makes reviews easier.

File: README (hunk 2 above)
Comment (efajardo) [reply README]: Actually it just needs a RHEL 6/7 machine.

**Comment by efajardo** [comment]
> A few minor suggestions but otherwise these changes are fine. One thing that's not immediately clear to me: are these like "light pilots" or are they payloads that run inside the pilot?
> 
> Also, in the future, please use a 120-char line limit. It makes reviews easier.

@brianhlin  I updated the requiements and explained this better.

File: README [hunk 6]
```diff
@@ -4,7 +4,8 @@ Use Case
 ========
 Rsv-gwms-tester is a package that is meant to do end-to-end regular testing of [GlideinWMS](https://glideinwms.fnal.gov/).
 More specifically it is mean to submit a set of HTCondor Vanilla universe jobs to the set of supported grid sites of a Virtual Organization (VO).
-It can be thought of as a [SAM-light](https://twiki.cern.ch/twiki/bin/view/CMSPublic/CompOpsSAMTests) set of grid tests.
+It can be thought of [HammerCloud-light](https://twiki.cern.ch/twiki/bin/view/Main/HammerCloud). The tests are payload jobs that run inside
+the pilto infrastructure
```
Comment (brianhlin) [inline README]:
```suggestion
the pilot infrastructure
```

Comment (brianhlin) [reply README]: Go ahead and merge this branch after fixing this tyupo

**Review by brianhlin — APPROVED** [review]

**Pull request merged**